scripts/verify-install.sh

```diff
@@ -158,29 +126,40 @@ for signature in $SIGNATURES; do
 done
 
 echo ""
-cd $TEMP_DIR || exit 1
 
 # Before we even look at the content of the manifest, we first want to make sure
 # the signatures actually sign that exact manifest.
 NUM_CHECKS=0
 for signature in $SIGNATURES; do
   echo "Verifying $signature"
-  if gpg --verify "$signature" "$MANIFEST" 2>&1 | grep -q "Good signature"; then
-    echo "Signature for $signature appears valid: "
-    gpg --verify "$signature" "$MANIFEST" 2>&1 | grep "using"
-  elif gpg --verify "$signature" 2>&1 | grep -q "No public key"; then
-    echo "Unable to verify signature $signature, no key available, skipping"
-    continue
+  VALID=0
+  for key in "${KEYS[@]}"; do
+    KEY_ID=$(echo $key | cut -d' ' -f1)
+    if GNUPGHOME="$TEMP_DIR/$KEY_ID" gpg --verify "$TEMP_DIR/$signature" "$TEMP_DIR/$MANIFEST" 2>&1 | grep -q "Good signature"; then
+      # We remove the key and signature to NOT count them more than once
+      # THIS IS CRUCIAL FOR VERIFICATION TO BE SECURE!!!
+      rm -rf "$TEMP_DIR/$KEY_ID" "$signature" || exit 1
```

Comment:
Great idea to count the number of distinct keys.

Reply:
I'm glad you like it, it didn't felt entirely clean but hopefully least foot-gunny.